src/Graviton/Rql/Visitor/MongoOdm.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Graviton\Rql\Visitor;
+
+use Graviton\Rql\AST\OperationInterface;
+use Doctrine\ODM\MongoDB\Query\Builder as QueryBuilder;
+
+class MongoOdm implements VisitorInterface
+{
+    /**
+     * @var QueryBuilder
+     */
+    private $queryBuilder;
+
+    /**
+     * @var string[]
+     */
+    private $queryOperations = array(
+        'and',
+        'or'
+    );
+
+    /**
+     * @var string[]
+     */
+    private $operationMap = array(
+        'eq' => 'equals',
+        'ne' => 'notEqual',
+        'lt' => 'lt',
+        'gt' => 'gt',
+        'lte' => 'lte',
+        'gte' => 'gte'
+);
+
+    /**
+     * @var string<string>
+     */
+    private $internalMethods = array(
+        'sort' => 'visitSort',
+        'like' => 'visitLike',
+        'limit' => 'visitLimit',
+    );
```

Comment:
Same as above, the Visitor should not be coupled to the Lexer. This is based on the AST.